.github/workflows/jdbc-tests-db-collation.yml

```diff
@@ -0,0 +1,151 @@
+        uses: ./.github/composite-actions/build-tds_fdw-extension
+
+      - name: Build vector Extension
+        id: build-vector-extension
+        if: always() &&  steps.build-tds_fdw-extension.outcome == 'success'
+        uses: ./.github/composite-actions/build-vector-extension
+
+      - name: Build PostGIS Extension
+        id: build-postgis-extension
+        if: always() &&  steps.build-vector-extension.outcome == 'success'
+        uses: ./.github/composite-actions/build-postgis-extension
+
+      - name: Install Extensions
+        id: install-extensions
+        if: always() && steps.build-postgis-extension.outcome == 'success'
+        uses: ./.github/composite-actions/install-extensions
+        with:
+          wal_level: logical
+
+      - name: Run JDBC Tests with DB level collation
+        id: jdbc
+        if: always() && steps.install-extensions.outcome == 'success'
+        timeout-minutes: 60
+        run: |
+          export PATH=/opt/mssql-tools/bin:$PATH
+          # Will uncomment it later when we plan to run JDBC tests for db level collation on a different db than master
+          # sqlcmd -S localhost -U "jdbc_user" -P 12345678 -i .github/scripts/db_collation.sql
+          export PATH=~/${{env.INSTALL_DIR}}/bin:$PATH
+          export PG_SRC=~/work/babelfish_extensions/postgresql_modified_for_babelfish
+          sudo ~/${{env.INSTALL_DIR}}/bin/psql -v ON_ERROR_STOP=1 -d jdbc_testdb -U runner -c "update sys.babelfish_sysdatabases set default_collation = '${{env.DATABASE_COLLATION_NAME}}' where name = 'master';"
+          sudo ~/${{env.INSTALL_DIR}}/bin/psql -v ON_ERROR_STOP=1 -d jdbc_testdb -U runner -c "select default_collation from sys.babelfish_sysdatabases where name = 'master';"
+          cd test/JDBC/
+          export databaseCollationName=${{env.DATABASE_COLLATION_NAME}}
+          mvn test
+          unset databaseCollationName
+
+      - name: Start secondary server
+        id: start-secondary
+        if: always() && steps.jdbc.outcome == 'success'
+        uses: ./.github/composite-actions/install-extensions
+        with:
+          psql_port: 5433
+          tsql_port: 8199
+          wal_level: logical
+
+      - name: Setup Publication and Subscription
+        id: setup-pub-sub
+        if: always() && steps.start-secondary.outcome == 'success'
+        run: |
+          ~/${{env.INSTALL_DIR}}/bin/psql -v ON_ERROR_STOP=1 -d jdbc_testdb -U runner -c "CREATE PUBLICATION my_pub;"
+          ~/${{env.INSTALL_DIR}}/bin/psql -v ON_ERROR_STOP=1 -d jdbc_testdb -U runner -p 5433 -c "CREATE SUBSCRIPTION my_sub CONNECTION 'host=localhost port=5432 user=jdbc_user dbname=jdbc_testdb password=12345678' PUBLICATION my_pub;"
+
+      - name: Run Replication Tests
+        id: replication
+        if: always() && steps.setup-pub-sub.outcome == 'success'
+        timeout-minutes: 60
+        uses: ./.github/composite-actions/run-jdbc-tests
+        with:
+          input_dir: 'replication'
```

Comment:
Are we testing these replication tests with non-default db collation ?

Reply:
Yes, the secondary server is being started whose configuration should be as same as primary server. We are connecting to the secondary server on port 5433 and updating the collation of master database to have identical configuration to primary server.